homeassistant/components/pterodactyl/api.py

```diff
@@ -88,20 +84,26 @@ async def async_get_data(self) -> dict[str, PterodactylData]:
         """Update the data from all Pterodactyl servers."""
         data = {}
 
-        if self.pterodactyl is None:
-            raise PterodactylNotInitializedError(
-                "Pterodactyl API is not initialized yet"
-            )
-
-        try:
-            for identifier in self.identifiers:
+        for identifier in self.identifiers:
+            try:
                 server = await self.hass.async_add_executor_job(
-                    self.pterodactyl.client.servers.get_server, identifier
+                    self.pterodactyl.client.servers.get_server,  # type: ignore[union-attr]
+                    identifier,
                 )
                 utilization = await self.hass.async_add_executor_job(
-                    self.pterodactyl.client.servers.get_server_utilization, identifier
+                    self.pterodactyl.client.servers.get_server_utilization,  # type: ignore[union-attr]
+                    identifier,
                 )
```

Comment:
Let's combine these 2 executor calls into one. Switching from the event loop to an executor job is expensive and it needs to wait a little, so the more we can group consecutive executor jobs, the faster we are